publish.py

```diff
@@ -62,100 +61,75 @@ class Formatter(argparse.RawDescriptionHelpFormatter, argparse.ArgumentDefaultsH
 
 
 class Repository(object):
-    RELEASE_TAG_PATTERN = re.compile(r"^v\d+(\.\d+){0,2}$")
+    RELEASE_TAG_PATTERN = re.compile(r"^v(0|[1-9]\d*)(\.(0|[1-9]\d*)){0,2}$")
 
     def __init__(self, options):
         self.repo = Repo(options.directory)
         self._force = options.force
-        self._current_tag = None
-
-    @property
-    def current_tag(self):
-        if self._current_tag:
-            return self._current_tag
-        try:
-            current_name = self.repo.git.describe(all=True)
-            self._current_tag = self.repo.rev_parse(current_name)
-            return self._current_tag
-        except (BadName, GitCommandError):
-            return None
-
-    @property
-    def version(self):
-        tag = self.current_tag
+        current_name = self.repo.git.describe(all=True)
+        print(f"Current name: {current_name}", file=sys.stderr)
+        current_tag = self.repo.rev_parse(current_name)
+        print(f"Current tag id: {current_tag}", file=sys.stderr)
         try:
-            return str(tag.tag)
-        except AttributeError:
-            return str(tag)
+          self.current_tag = current_tag.tag
+          print(f"Current tag: {self.current_tag}", file=sys.stderr)
+        except (AttributeError, BadName, GitCommandError):
+          print("Not an annotated tag!", file=sys.stderr)
+          exit(4)
```

Comment:
Since GitCommandError is caught here, I think you might've meant to include the git calls inside the try block.